pgjdbc/src/main/java/org/postgresql/jdbc/PgConnection.java

```diff
@@ -1369,7 +1369,7 @@ public boolean isValid(int timeout) throws SQLException {
         // "current transaction aborted", assume the connection is up and running
         return true;
       }
-      LOGGER.log(Level.WARNING, GT.tr("Validating connection."), e);
+      LOGGER.log(Level.FINE, GT.tr("Validating connection."), e);
```

Comment:
-1 The exception is not re-thrown, so hiding it by default might make "root cause analysis" harder.

Reply:
Yes, the exception is not re-thrown but the method returns a boolean indicating if the connection is Valid, the point of the method is not get the "root cause", is to validate the connection.

For instance if a connection pool every time it tries to borrow a connection check if is valid, this will flood the log for no reason.